magefile.go

```diff
@@ -61,79 +70,177 @@ var Aliases = map[string]interface{}{
 type All mg.Namespace
 
 func (All) Check(ctx context.Context) {
-	mg.SerialCtxDeps(ctx,
+	mg.SerialCtxDeps(
+		ctx,
 		Check.Tidy,
 		Check.Verify,
 		Check.Lint,
 	)
 }
 
-func (All) Clean() {
-	mg.Deps(
+func (All) Clean(ctx context.Context) {
+	mg.CtxDeps(
+		ctx,
 		Build.CleanCLI,
 		Build.CleanOperator,
 		Test.Clean,
 	)
 }
 
-func (All) Test() {
-	mg.Deps(
+func (All) Test(ctx context.Context) {
+	mg.CtxDeps(
+		ctx,
 		Test.Unit,
 		Test.Integration,
 	)
 }
 
 type Check mg.Namespace
 
-func (Check) Tidy() error {
-	return sh.Run(mg.GoCmd(), "mod", "tidy")
+func (Check) Tidy(ctx context.Context) error {
+	tidy := gocmd(
+		command.WithArgs{"mod", "tidy"},
+		command.WithConsoleOut(mg.Verbose()),
+		command.WithContext{Context: ctx},
+	)
+
+	if err := tidy.Run(); err != nil {
+		return fmt.Errorf("starting tidy: %w", err)
+	}
+
+	if tidy.Success() {
+		return nil
+	}
+
+	return fmt.Errorf("running tidy: %w", tidy.Error())
 }
 
-func (Check) Verify() error {
-	return sh.Run(mg.GoCmd(), "mod", "verify")
+func (Check) Verify(ctx context.Context) error {
+	verify := gocmd(
+		command.WithArgs{"mod", "verify"},
+		command.WithConsoleOut(mg.Verbose()),
+		command.WithContext{Context: ctx},
+	)
+
+	if err := verify.Run(); err != nil {
+		return fmt.Errorf("starting verification: %w", err)
+	}
+
+	if verify.Success() {
+		return nil
+	}
+
+	return fmt.Errorf("running verification: %w", verify.Error())
 }
 
 func (Check) Lint(ctx context.Context) error {
 	mg.CtxDeps(ctx, Deps.UpdateGolangCILint)
 
-	return sh.Run(filepath.Join(_depBin, "golangci-lint"), "run",
-		"--timeout=10m",
-		"-E", "unused,gofmt,goimports,gosimple,staticcheck",
-		"--skip-dirs-use-default",
-		"--verbose",
+	lint := golangci(
+		command.WithArgs{"run",
+			"--timeout=10m",
+			"-E", "unused,gofmt,goimports,gosimple,staticcheck",
+			"--skip-dirs-use-default",
+			"--verbose",
+		},
+		command.WithContext{Context: ctx},
 	)
+
+	if err := lint.Run(); err != nil {
+		return fmt.Errorf("starting linter: %w", err)
+	}
+
+	fmt.Fprint(os.Stdout, lint.CombinedOutput())
+
+	if lint.Success() {
+		return nil
+	}
+
+	return fmt.Errorf("running linter: %w", lint.Error())
 }
 
+var golangci = command.NewCommandAlias(filepath.Join(_depBin, "golangci-lint"))
+
 type Test mg.Namespace
 
-func (Test) Unit() error {
-	return sh.RunWith(map[string]string{
-		"CGO_CFLAGS": "-DSQLITE_ENABLE_JSON1",
-	}, mg.GoCmd(), "test", "-count=1", "-race", "-timeout", "15m",
-		"./api...",
-		"./cmd...",
-		"./internal...",
-		"./pkg...",
+func (Test) Unit(ctx context.Context) error {
+	test := gocmd(
+		command.WithCurrentEnv(true),
+		command.WithEnv{
+			"CGO_CFLAGS": "-DSQLITE_ENABLE_JSON1",
+		},
+		command.WithArgs{
+			"test", "-count=1", "-race", "-timeout", "15m",
+			"./api...",
+			"./cmd...",
+			"./internal...",
+			"./pkg...",
```

Comment:
Is there a way to reverse that logic. If we can simply exclude `integration` we don't have to maintain that list for new packages/dirs?